x/ibc/04-channel/exported/exported.go

```diff
@@ -0,0 +1,16 @@
+package exported
+
+type PacketI interface {
+	Sequence() uint64
+	TimeoutHeight() uint64
+	SourcePort() string
+	SourceChannel() string
+	DestPort() string
+	DestChannel() string
+	Data() []byte
+
+	// // Non ICS04 interface functions
```

Comment:
What is the purpose of these additional functions and do they need to be implemented?

Reply:
I have the same question. Also a packet should really be a concrete structure, not an interface.